### PR TITLE
Add high order flux correction for Cartesian meshes

### PR DESCRIPTION
**Pull request by nilsdeppe**
### Code review checklist

- [ ] The code is documented and the documentation renders correctly. Run
  `make doc` to generate the documentation locally into `BUILD_DIR/docs/html`.
  Then open `index.html`.
- [ ] The code follows the stylistic and code quality guidelines listed in the
  [code review guide](https://spectre-code.org/code_review_guide.html).
- [ ] The PR lists upgrade instructions and is labeled `bugfix` or
  `new feature` if appropriate.

### CODE REVIEW

**Comment by nilsdeppe** [comment]
I need to make some updates to this after getting the FD derivatives working in an evolution.

**Comment by nilsdeppe** [comment]
Found some minus sign bugs that weren't caught by the test (because my test wasn't quite right). Will fix and push.

File: src/NumericalAlgorithms/FiniteDifference/HighOrderFluxCorrection.hpp [hunk 1]
```diff
@@ -0,0 +1,336 @@
...
+namespace fd {
+/// @{
+/*!
+ * \brief Computes a high-order boundary correction $G$ at the FD interface.
+ *
+ * The correction to the second-order boundary correction is given by
+ *
+ * \f{align*}{
+ *  G=G^{(2)}-G^{(4)}+G^{(6)}-G^{(8)}+G^{(10)},
+ *\f}
```
Comment (kidder) [inline src/NumericalAlgorithms/FiniteDifference/HighOrderFluxCorrection.hpp]: there is an `*` at the end of this line when I render it.  Can you take a look?  might be the lack of a space between `*` and `\f}` on line 30?

File: src/NumericalAlgorithms/FiniteDifference/HighOrderFluxCorrection.hpp [hunk 2]
```diff
@@ -0,0 +1,336 @@
...
+ *
+ *\f{align*}{
+ * G^{(4)}_{j+1/2}&=\frac{1}{6}\left(G_j -2 G^{(2)} +
+ *                         G_{j+1}\right), \\
+ * G^{(6)}_{j+1/2}&=\frac{1}{180}\left(G_{j-1} - 9 G_j + 16 G^{(2)}
+ *                         -9 G_{j+1} + G_{j+2}\right), \\
+ * G^{(8)}_{j+1/2}&=\frac{1}{2100}\left(G_{j-2} - \frac{25}{3} G_{j-1}
+ *                         + 50 G_j - \frac{256}{3} G^{(2)} + 50 G_{j+1}
+ *                         - \frac{25}{3} G_{j+2} +G_{j+3}\right), \\
+ * G^{(10)}_{j+1/2}&=\frac{1}{17640}
```
Comment (kidder) [inline src/NumericalAlgorithms/FiniteDifference/HighOrderFluxCorrection.hpp:42]: the previous three are symmetric, why isn't this one?

File: src/NumericalAlgorithms/FiniteDifference/HighOrderFluxCorrection.hpp [hunk 3]
```diff
@@ -0,0 +1,336 @@
...
+ * This is a generalization of the correction presented in \cite CHEN2016604.
+ *
+ * This high-order flux can be fed into a flux limiter, e.g. to guarantee
+ * positivity.
+ *
+ * \note This implementation should be profiled and optimized.
+ *
+ * \warning This documentation is for the general case. In the restricted
+ * Cartesian case we use the cell-centered flux as opposed to `G^{(4)}`, which
+ * differ by a minus sign. This amounts a minus sign change in front of the
```
Comment (kidder) [inline src/NumericalAlgorithms/FiniteDifference/HighOrderFluxCorrection.hpp]: differs

Comment (kidder) [reply src/NumericalAlgorithms/FiniteDifference/HighOrderFluxCorrection.hpp]: amounts to

File: src/NumericalAlgorithms/FiniteDifference/HighOrderFluxCorrection.hpp [hunk 4]
```diff
@@ -0,0 +1,336 @@
...
+ * $G^{(k)}$ terms in computing $G$ for $k>2$, and also a sign change in from
```
Comment (kidder) [inline src/NumericalAlgorithms/FiniteDifference/HighOrderFluxCorrection.hpp]: in front

File: src/NumericalAlgorithms/FiniteDifference/HighOrderFluxCorrection.hpp [hunk 5]
```diff
@@ -0,0 +1,336 @@
...
+                                    correction_width + 2)) +
+                     49.0 * (gsl::at(cell_centered_fluxes_for_stencil,
+                                     correction_width - 2) +
+                             gsl::at(cell_centered_fluxes_for_stencil,
+                                     correction_width + 1)) -
+                     245.0 * (gsl::at(cell_centered_fluxes_for_stencil,
+                                      correction_width - 1) +
+                              gsl::at(cell_centered_fluxes_for_stencil,
+                                      correction_width)) -
+                     409.6 * second_order_var_correction[storage_index]
```
Comment (kidder) [inline src/NumericalAlgorithms/FiniteDifference/HighOrderFluxCorrection.hpp:216]: okay this doesn't match the doc, but is symmetric...

**Review by kidder — CHANGES_REQUESTED** [review]
you can squash

File: src/NumericalAlgorithms/FiniteDifference/HighOrderFluxCorrection.hpp (hunk 5 above)
Comment (nilsdeppe) [reply src/NumericalAlgorithms/FiniteDifference/HighOrderFluxCorrection.hpp:216]: The docs was just wrong... I've fixed it in a fixup

**Review by kidder — COMMENTED** [review]
squash

**Comment by nilsdeppe** [comment]
Squashed. Thanks!

**Review by kidder — APPROVED** [review]

**Pull request merged**